S2VX.Game/Story/Note/GameNote.cs

```diff
@@ -52,40 +54,72 @@ public class GameNote : S2VXNote {
             Delete();
         }
 
-        // Notes are clickable if they are visible on screen and not missed
+        // Notes are clickable if they are visible on screen, not missed, and is the earliest note
         private bool IsClickable() {
             var time = Time.Current;
-            // Limit timing error to be +/- MissThreshold (though it will never be >= MissThreshold since RecordMiss would have already run) 
+            // Limit timing error to be +/- MissThreshold (though it will never be >= MissThreshold since RecordMiss would have already run)
             TimingError = (int)Math.Round(Math.Clamp(time - EndTime, -MissThreshold, MissThreshold));
-            return TimingError <= MissThreshold && Alpha > 0;
+            var inMissThreshold = TimingError <= MissThreshold && Alpha > 0;
+            var earliestNote = (GameNote)Story.Notes.Children[^1];
+            var isEarliestNote = earliestNote == this;
+            return inMissThreshold && isEarliestNote;
+        }
+
+        private void ClickNote() {
+            ScoreInfo.AddScore(Math.Abs(TimingError));
+            Delete();
+            BlockClicking = true;
+        }
+
+        // Unblocks clicking on the last call of OnMouseDown / OnKeyDown
+        private void CheckAndResetClicking() {
+            if (Story.Notes.Children.Count > 0 && this == Story.Notes.Children[0]) {
```

Comment:
Instead of `Children[^1]` and `Children[0]`, you might want to use the more idiomatic `Children.Last()` and `Children.First()` functions. I don't think you need the `Children.Count > 0` check either because if this function triggers there should always be a `Note`. If you'd like to be safe (and also my preference), you can use the `Children.LastOrDefault()` and `Children.FirstOrDefault()` functions. These functions will return `null` if First or Last do not exist, and you can compare that directly with `this`.